### PR TITLE
Panwang/taskruns

### PR DESCRIPTION
**Pull request by pan-wang**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by anuchandy — APPROVED** [review]
Looks good from SDK side. Requested ARM review as it adds new APIs.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

**Comment by yungezz** [comment]
hi @pan-wang could you pls address comments ?

**Comment by pan-wang** [comment]
Thank so much for remaindering.
We are working on twitting the change for better ARM deployment experience. Will address all feedbacks soon once we land on final change.

Thanks

Pan
________________________________

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (majastrz) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: Yes, you need to add schema for the error.

---
In reply to: [324001409](https://github.com/Azure/azure-rest-api-specs/pull/7123#discussion_r324001409) [](ancestors = 324001409)

**Review by majastrz — CHANGES_REQUESTED** [review]
@KrisBash's comments have not been addressed and are blocking approval. Please fix.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (majastrz) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: Why are you removing the "default" error response? Adding it was fine, but you also need to specify the schema. There are many examples of swaggers doing it. Here's one: [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/resources.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/resources.json)

---
In reply to: [343912014](https://github.com/Azure/azure-rest-api-specs/pull/7123#discussion_r343912014) [](ancestors = 343912014,324001409)

Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @majastrz We didn't have default response in our other RP. It is better to make this one be consistent with others. If there's any new request, we would like to create a separate work item and fix together. //cc: @pan-wang

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: Same here, to be consistent, we just remove default responses.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (northtyphoon) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: Kristopher, we have been using string for type for several versions. Since it's not related to this PR, are you OK we investigate a little bit more and start another PR to address it?

---
In reply to: [324002726](https://github.com/Azure/azure-rest-api-specs/pull/7123#discussion_r324002726) [](ancestors = 324002726)

**Comment by huanwu** [comment]
@krishbabu Could we move forward for this PR? We would like to file a separate work item to solve the debt per discussion above. And there're two Ignore Test-Only CI fail. We can just ignore, right?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TaskRunsCreate.json
Comment (majastrz) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TaskRunsCreate.json]: Are any of these values secrets?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (majastrz) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: I think that should be fine.

---
In reply to: [344367022](https://github.com/Azure/azure-rest-api-specs/pull/7123#discussion_r344367022) [](ancestors = 344367022)

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TaskRunsCreate.json
Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TaskRunsCreate.json]: no

**Comment by majastrz** [comment]
> @krishbabu Could we move forward for this PR? We would like to file a separate work item to solve the debt per discussion above. And there're two Ignore Test-Only CI fail. We can just ignore, right?

I think you meant @KrisBash. For the CI failures, @anuchandy will need to weigh as the SDK team owns these checks.

**Review by majastrz — APPROVED** [review]
Looks good to me. I'll leave it for @KrisBash to do the final sign off since his comments are still remaining.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @majastrz After review the doc, we add back the default error response with the schema.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: The error schema was added.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (ravbhatnagar) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: this is not the right schema for error. sample here - https://github.com/Azure/azure-rest-api-specs/blob/master/specification/batch/resource-manager/Microsoft.Batch/stable/2019-08-01/BatchManagement.json#L105

**Review by ravbhatnagar — COMMENTED** [review]
@pan-wang - one comment, please take a look. ALso some from Kris have not been addressed.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (huanwu) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @ravbhatnagar Just submitted a new commit to solve the problem

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Comment by huanwu** [comment]
@majastrz ARM signed off this change. Could you continue merge the manifest?

**Comment by huanwu** [comment]
@anuchandy Could you merge the PR? The manifest update was blocked now.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
